test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -431,6 +432,40 @@ private StringDescription descriptionOf(Matcher<?> matcher) {
         return this;
     }
 
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Object exceptionDetails) {
+        return expectExceptionDetails(CoreMatchers.equalTo(exceptionDetails));
+    }
+
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Class<?> exceptionDetails) {
+        return expectExceptionDetails(instanceOf(exceptionDetails));
+    }
+
+    @Override
+    public ResultValidator<T> expectExceptionDetails(Matcher<?> exceptionDetailsMatcher) {
+        Object actualDetails = HandlerExecutionException.resolveDetails(actualException).orElse(null);
+        if (exceptionDetailsMatcher == null) {
+            StringDescription emptyMatcherDescription = new StringDescription(
+                    new StringBuilder("Given exception details matcher is null!"));
+            reporter.reportWrongExceptionDetails(actualDetails, emptyMatcherDescription);
+            return this;
+        }
+        if (actualException == null) {
+            StringDescription description = new StringDescription(new StringBuilder(
+                    "an exception with details matching "));
+            exceptionDetailsMatcher.describeTo(description);
+            reporter.reportUnexpectedReturnValue(actualReturnValue.getPayload(), description);
```

Comment:
I added the additional if-block as @smcvb suggested.

The error is reported using the same reporter method than in `expectException(Matcher)` but with an additional prefix for the description. Given a `containsString("some string")` matcher, it would report `an exception with details matching a string containing "some string"`. I am not used to manipulate matchers and their description, thus I am not sure this wording would work for every matcher description. Thoughts ?

Reply:
In all honesty, I'm not 100% certain either.
Off the top of my head, I think we should be good with this, though.

If the text turns out to be malformed in some scenarios, we'll just provide a PR to fix that once and if it occurs.